crm_partner_contact_role/models/crm_lead.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    role_ids = fields.Many2many(
+        comodel_name="res.partner.role",
+        compute="_compute_role",
```

Comment:
Would there be an issue if we just made it a related field?

Reply:
Yes. It will when we want to create new customer with filling role_ids and contact name from `pipeline` without assigning `partner_id`.